Editor/ObjectMapping/ObjectMappingBuilder.cs

```diff
@@ -41,26 +52,86 @@ public ObjectMappingBuilder(GameObject rootObject)
+                        FirstPersonFlag.FirstPersonOnly => VrmFirstPersonFlag.FirstPersonOnly,
+                        _ => throw new ArgumentOutOfRangeException()
+                    };
+                }
+            }
+#endif
+
+#if AAO_VRM1
+            if (rootObject.TryGetComponent<Vrm10Instance>(out var vrm10Instance))
+            {
+                foreach (var renderer in vrm10Instance.Vrm.FirstPerson.Renderers)
+                {
+                    GetComponentInfo(renderer.GetRenderer(rootObject.transform)).VrmFirstPersonFlag = renderer.FirstPersonFlag switch
+                    {
+                        FirstPersonType.auto => VrmFirstPersonFlag.Auto,
+                        FirstPersonType.both => VrmFirstPersonFlag.Both,
+                        FirstPersonType.thirdPersonOnly => VrmFirstPersonFlag.ThirdPersonOnly,
+                        FirstPersonType.firstPersonOnly => VrmFirstPersonFlag.FirstPersonOnly,
+                        _ => throw new ArgumentOutOfRangeException()
+                    };
+                }
+            }
+#endif
         }
 
         public void RecordMergeComponent<T>(T from, T mergeTo) where T: Component
         {
+            var newMergeToInfo = new BuildingComponentInfo(mergeTo);
+
+#if AAO_VRM0 || AAO_VRM1
+            newMergeToInfo.VrmFirstPersonFlag = MergeVrmFirstPersonFlags(GetVrmFirstPersonFlag(from), GetVrmFirstPersonFlag(mergeTo));
+#endif
+
             if (!_componentInfos.TryGetValue(mergeTo.GetInstanceID(), out var mergeToInfo))
             {
-                var newMergeToInfo = new BuildingComponentInfo(mergeTo);
                 _originalComponentInfos.Add(mergeTo.GetInstanceID(), newMergeToInfo);
                 _componentInfos.Add(mergeTo.GetInstanceID(), newMergeToInfo);
                 GetComponentInfo(from).MergedTo(newMergeToInfo);
             }
             else
             {
-                var newMergeToInfo = new BuildingComponentInfo(mergeTo);
                 _componentInfos[mergeTo.GetInstanceID()]= newMergeToInfo;
                 mergeToInfo.MergedTo(newMergeToInfo);
                 GetComponentInfo(from).MergedTo(newMergeToInfo);
             }
         }
 
+#if AAO_VRM0 || AAO_VRM1
+        VrmFirstPersonFlag? MergeVrmFirstPersonFlags(VrmFirstPersonFlag? from, VrmFirstPersonFlag? to)
+        {
+            if (!from.HasValue)
+            {
+                return to;
+            }
+            if (!to.HasValue)
+            {
+                return from;
+            }
+            if (from == to)
+            {
+                return from;
+            }
+            
+            var mergedFirstPersonFlag = from == VrmFirstPersonFlag.Both || to == VrmFirstPersonFlag.Both ? VrmFirstPersonFlag.Both : VrmFirstPersonFlag.Auto; 
+            BuildLog.LogWarning("MergeSkinnedMesh:warning:VRM:FirstPersonFlagsMismatch", mergedFirstPersonFlag.ToString());
+            return mergedFirstPersonFlag;
+        }
```

Comment:
```suggestion
        {
            switch (from, to)
            {
                case (null, var other): return other;
                case (var other, null): return other; 
                case ({ } fromFlag, { } toFlag) when fromFlag == toFlag: return fromFlag;
                case ({ } fromFlag, { } toFlag):
                {
                    var mergedFirstPersonFlag = fromFlag == VrmFirstPersonFlag.Both || toFlag == VrmFirstPersonFlag.Both
                        ? VrmFirstPersonFlag.Both : VrmFirstPersonFlag.Auto; 
                    BuildLog.LogWarning("MergeSkinnedMesh:warning:VRM:FirstPersonFlagsMismatch", mergedFirstPersonFlag.ToString());
                    return mergedFirstPersonFlag;
                }
            }
        }
```

HasValueではなくswitchにしたいです